include/cantera/thermo/PengRobinson.h

```diff
@@ -0,0 +1,359 @@
+     * standard state as an ideal gas at the current temperature and pressure of
+     * the solution. The activities are based on this standard state.
+     *
+     * @param ac Output vector of activity coefficients. Length: m_kk.
+     */
+    virtual void getActivityCoefficients(double* ac) const;
+
+    /// @name  Partial Molar Properties of the Solution
+    //@{
+
+    //! Get the array of non-dimensional species chemical potentials.
+    //! These are partial molar Gibbs free energies.
+    /*!
+     * \f$ \mu_k / \hat R T \f$.
+     * Units: unitless
+     *
+     * We close the loop on this function here calling getChemPotentials() and
+     * then dividing by RT. No need for child classes to handle.
+     *
+     * @param mu    Output vector of non-dimensional species chemical potentials
+     *              Length: m_kk.
+     */
+    
+    virtual void getChemPotentials(double* mu) const;
+    virtual void getPartialMolarEnthalpies(double* hbar) const;
+    virtual void getPartialMolarEntropies(double* sbar) const;
+    virtual void getPartialMolarIntEnergies(double* ubar) const;
+    virtual void getPartialMolarCp(double* cpbar) const;
+    virtual void getPartialMolarVolumes(double* vbar) const;
+
+    //! Calculate the temperature dependent interaction parameter alpha needed for P-R EoS
+    /*
+    *  The temperature dependent parameter in P-R EoS is calculated as
+    *       \alpha = [1 + \kappa(1 - sqrt{T/T_crit}]^2
+    *  kappa is a function calulated based on the accentric factor.
+    *  Units: unitless
+	* a and b are species-specific coefficients used in P-R EoS, w is the acentric factor.
+    */
+    virtual void calculateAlpha(const std::string& species, double a, double b, double w);
+    //@}
+    /// @name Critical State Properties.
+    //@{
+
+    virtual double critTemperature() const;
+    virtual double critPressure() const;
+    virtual double critVolume() const;
+    virtual double critCompressibility() const;
+    virtual double critDensity() const;
+    virtual double speciesCritTemperature(double a, double b) const;
+
+public:
+    //@}
+    //! @name Initialization Methods - For Internal use
+    /*!
+     * The following methods are used in the process of constructing
+     * the phase and setting its parameters from a specification in an
+     * input file. They are not normally used in application programs.
+     * To see how they are used, see importPhase().
+     */
+    //@{
+
+    virtual bool addSpecies(shared_ptr<Species> spec);
+    virtual void initThermo();
+
+    //! Retrieve a and b coefficients by looking up tabulated critical parameters
+    /*!
+    *  If pureFluidParameters are not provided for any species in the phase,
+    *  consult the critical properties tabulated in data/inputs/critProperties.xml.
+    *  If the species is found there, calculate pure fluid parameters a_k and b_k as:
+    *  \f[ a_k = 0.4278*R**2*T_c^2/P_c \f]

```

Comment:
Please don't use `*` for multiplication, and is this `R^2`?

Reply:
Corrected.